src/app/[locale]/(auth)/dashboard/handlebars-playground/handlebars-playground.tsx

```diff
@@ -0,0 +1,158 @@
+import { fetchTemplateById, saveTemplate, updateTemplate } from '@/libs/actions/templates';
+import { TemplateType } from '@/types/Template';
+
+const HandlebarsPlayground = () => {
+  const [template, setTemplate] = useState<string>(''); // Handlebars template
+  const [inputData, setInputData] = useState<string>(''); // JSON data for the template
+  const [output, setOutput] = useState<string>(''); // Rendered HTML
+  const [error, setError] = useState<string | null>(null); // Error messages
+
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const templateId = searchParams.get('templateId'); // Get templateId from query params
+
+  const handleRender = () => {
+    setError(null);
+
+    try {
+      // Compile the Handlebars template
+      const compiledTemplate = Handlebars.compile(template);
+
+      // Parse the JSON input
+      const context = inputData.trim() ? JSON.parse(inputData) : {};
+
+      // Render the template with the context
+      const result = compiledTemplate(context);
+      setOutput(result);
+    } catch (err: any) {
+      setError(`Failed to render template: ${err.message}`);
+    }
+  };
+
+  // Automatically update the preview when template or JSON changes
+  useEffect(() => {
+    if (template || inputData) {
+      handleRender();
+    }
+  }, [template, inputData]);
+
+  useEffect(() => {
+    const loadTemplateData = async () => {
+      if (templateId) {
+        const response = await fetchTemplateById(templateId);
+        if (!response.success) {
+          return;
+        }
+        // Load the template data into the editor
+        setTemplate(response.data?.templateContent as string);
+        setInputData(response.data?.templateSampleData as string);
+      }
+    };
+    loadTemplateData();
+    handleRender();
+  }, [templateId]);
+
+  const handleSave = async () => {
+    if (templateId) {
+      try {
+        const response = await updateTemplate({
+          templateId,
+          description: 'My Handlbars Template',
+          templateContent: template,
+          templateStyle: '',
+          templateSampleData: JSON.stringify(inputData),
+        });
+
+        if (response.success) {
+          router.push('/dashboard');
+        } else {
+          throw new Error(response.error);
+        }
+      } catch (error: any) {
+        console.error('Error saving template:', error);
+      }
+    } else {
+      try {
+        // Sample input data for the server action
+        const assets = ['https://example.com/asset1.png', 'https://example.com/asset2.png'];
+
+        const response = await saveTemplate({
+          description: 'My Handlebars Template',
+          userId: 'c9ded72d-4cae-4fab-b86c-a084ec7f2ecc',
+          templateContent: template,
+          templateSampleData: JSON.stringify(inputData),
+          templateStyle: '',
+          assets: JSON.stringify(assets),
+          templateType: TemplateType.HANDLBARS_TEMPLATE,
```

Comment:
templatetype should be determined based on the button selected by the user on UI right